pumpkin-lib/src/branching/branchers/alternating_brancher.rs

```diff
@@ -143,8 +155,35 @@ impl<OtherBrancher: Brancher> Brancher for AlternatingBrancher<OtherBrancher> {
 
     fn on_restart(&mut self) {
         if let AlternatingStrategy::EveryRestart = self.strategy {
-            // Switch whenever a restart occurs
-            self.toggle_brancher()
+            // We have considered a restart and we should switch
+            self.has_considered_restart = true;
+        }
+    }
+
+    fn is_restart_pointless(&mut self) -> bool {
+        match self.strategy {
+            AlternatingStrategy::EveryRestart => {
+                // In the case of the `EveryRestart` strategy, note that we switch to the other
+                // strategy and then the restart is performed so we check whether restarting for the
+                // other brancher is pointless
+                let is_restart_pointless = if self.is_using_default_brancher {
+                    self.other_brancher.is_restart_pointless()
+                } else {
+                    self.default_brancher.is_restart_pointless()
+                };
+                // We indicate that we have considered a restart, this can then be used by the
+                // EveryRestart AlternatingStrategy to determine when to switch
+                self.has_considered_restart = true;
```

Comment:
If we set the flag before the if statement, we can inline the `is_restart_pointless` variable